x-pack/plugins/synthetics/public/legacy_uptime/components/monitor_management/manage_locations/locations_list.tsx

```diff
@@ -0,0 +1,139 @@
+import { selectAgentPolicies } from '../../../state/private_locations';
+
+export const PrivateLocationsList = ({
+  privateLocations,
+  onSubmit,
+  loading,
+  onDelete,
+}: {
+  privateLocations: PrivateLocation[];
+  loading: boolean;
+  onSubmit: (location: PrivateLocation) => void;
+  onDelete: (id: string) => void;
+}) => {
+  const { basePath } = useUptimeSettingsContext();
+
+  const { data: policies } = useSelector(selectAgentPolicies);
+
+  const { locations } = useLocationMonitors();
+
+  if (loading) {
+    return <EuiLoadingSpinner />;
+  }
+
+  return (
+    <Wrapper>
+      {privateLocations.map((location) => {
+        const monCount = locations?.find((l) => l.id === location.id)?.count ?? 0;
+        const canDelete = monCount === 0;
+        const policy = policies?.items.find((policyT) => policyT.id === location.policyHostId);
+        return (
+          <div key={location.id}>
+            <EuiAccordion
+              id={location.id}
+              element="fieldset"
+              className="euiAccordionForm"
+              buttonClassName="euiAccordionForm__button"
+              buttonContent={
+                <div>
+                  <EuiFlexGroup gutterSize="s" alignItems="center" responsive={false}>
+                    <EuiFlexItem grow={false}>
+                      <EuiIcon type="visMapCoordinate" size="m" />
+                    </EuiFlexItem>
+
+                    <EuiFlexItem>
+                      <EuiTitle size="xs" className="eui-textNoWrap">
+                        <h3>{location.name}</h3>
+                      </EuiTitle>
+                    </EuiFlexItem>
+
+                    <EuiFlexItem>
+                      <EuiText size="xs">
+                        <EuiTextColor color="subdued">Running monitors: {monCount}</EuiTextColor>
+                      </EuiText>
+                    </EuiFlexItem>
+                  </EuiFlexGroup>
+
+                  <EuiText size="s">
+                    <p>
+                      <EuiTextColor color="subdued">
+                        Host policy:{' '}
+                        {policy ? (
+                          <EuiLink href={`${basePath}/app/fleet/policies/${location.policyHostId}`}>
+                            {policy?.name}
+                          </EuiLink>
+                        ) : (
+                          <EuiText color="danger" size="s" className="eui-displayInline">
+                            Policy is deleted
+                          </EuiText>
+                        )}
+                      </EuiTextColor>
+                    </p>
+                  </EuiText>
+                </div>
+              }
+              extraAction={
+                <EuiToolTip
+                  title={
+                    canDelete
+                      ? 'Delete location'
+                      : `it can't be deleted, because it has ${monCount} monitors running. Please delete them before you can delete this location`
```

Comment:
```suggestion
                      : `This location cannot be deleted because it has ${monCount} monitors running. Please remove this location from your monitors before deleting this location.
```
Content suggestion, but I think we should automate this in the future.